pkg/bloomcompactor/config.go

```diff
@@ -0,0 +1,36 @@
+package bloomcompactor
+
+import (
+	"flag"
+
+	"github.com/grafana/loki/pkg/util"
+)
+
+// Config configures the bloom-compactor component.
+type Config struct {
+	// Ring configures the ring store used to save and retrieve the different Bloom-Compactor instances.
+	// In case it isn't explicitly set, it follows the same behavior of the other rings (ex: using the common configuration
+	// section and the ingester configuration by default).
+	Ring RingCfg `yaml:"ring,omitempty" doc:"description=Defines the ring to be used by the bloom-compactor servers. In case this isn't configured, this block supports inheriting configuration from the common ring section."`
+	// Enabled configures whether bloom-compactors should be used to compact index values into bloomfilters
+	Enabled bool `yaml:"enabled"`
+}
+
+// RegisterFlags registers flags for the Bloom-Compactor configuration.
+func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	cfg.Ring.RegisterFlags("bloom-compactor.", "collectors/", f)
+	f.BoolVar(&cfg.Enabled, "bloom-compactor.enabled", false, "Flag to enable or disable the usage of the bloom-compactor component.")
+}
+
+// RingCfg is a wrapper for our internally used ring configuration plus the replication factor.
+type RingCfg struct {
```

Comment:
I think we can just use `util.RingConfig` rather than redefining another type here. I also don't think we'll need replication factors in the bloom compactors.

Reply:
we all decided to extract ring code in a separate PR for all these new services.